<launch>
	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.5" />
		<param name="latch" type="bool" value="false"/>

		<param name="frame_id" type="string" value="map" />
		<param name="base_frame_id" type="string" value="map" />

		<param name="sensor_model/max_range" value="50.0" />

		<!--
		<param name="filter_ground" type="bool" value="true" />
		<param name="ground_filter/distance" type="double" value="0.5" />
    	<param name="ground_filter/plane_distance" type="double" value="0.7" />
		<param name="ground_filter/angle" type="double" value="1" />
		-->
		<param name="occupancy_min_z" type="double" value="3.0" />
    	<param name="occupancy_max_z" type="double" value="200.0" />

		<remap from="cloud_in" to="/carla_node/lidar_point_cloud" />
	</node>

	<!-- Carla Multi-LiDAR Merger Node -->
    <node pkg="mapper" type="carla_merged_lidar_publisher.py" name="carla_merged_lidar_publisher" output="screen">
        <param name="use_sim_time" value="true" />
    </node>
</launch>